views/setPassword.ejs

```diff
@@ -0,0 +1,88 @@
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/open-fonts@1.1.1/fonts/inter.min.css">
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@exampledev/new.css@1.1.2/new.min.css">
+
+<style>
+    * {
+        text-align: center;
+    }
+    input {
+        width: 400px;
+        height: 40px;
+        margin: 5px;
+    }
+    button {
+        width: 400px;
+        height: 40px;
+        margin: 10px;
+    }
+    .card {
+        width: 450px;
+        margin: auto;
+    }
+    .subtitle {
+        margin: 10px;
+    }
+    .body {
+        margin: 10px;
+    }
+</style>
+
+<main>
+    <div class="card">
+        <div class="head">
+            <h1 class="title">Learn Databases</h1>
+        </div>
+        <div class="body">
+            <div class="subtitle">Set your password</div>
+            <input class="password1" placeholder="create new password" type="password"/><br>
+            <input class="password2" placeholder="confirm password" type="password"/>
+            <button class="submit">Submit</button>
+        </div>
+    </div>
+</main>
+
+<script>
+    const path = window.location.pathname.split('/')
+    const token = path.pop()
+    console.log('path is', path)
+    console.log('toekn is', token)
+
+    const password1 = document.querySelector('.password1')
+    const password2 = document.querySelector('.password2')
+    const submit = document.querySelector('.submit')
+
+    submit.addEventListener('click', () => {
+        if (password1.value !== password2.value) {
+            return alert("Passwords do not match. Please check again.")
+        }
+        fetch('/api/passwordReset', {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                token: token,
+                password: password1.value
+            })
+        }).then(r => r.json()).then((r) => {
+            if (r.error) return alert(r.error.message)
+            fetch('/api/session', {
```

Comment:
`return fetch(....)` You should not have nested promises. Promises goal is to not have these nested promises.

Isn't it better to do `location.href = '/'`?

Reply:
I think it's good to automatically log in the user when they set there password (I have lots of memory thinking that it's annoying that I have to tell the whole user information again just after I told them)
I can do this way
```
submit.addEventListener('click', asyc () => {
  try {
    fetch('/api/passwordReset', {...}).then(...)
  } catch {
  return
  }
  try {
    fetch('/api/session', {...}).then((r) => { location.href='/setDBPassword' })
  } catch {
  ...
  }
})
```